ansible/vars_plugins/load_secrets.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DECRYPT_TOOL = "dotgpg"
+
+import os
+import sys
+import tempfile
+import errno
+import subprocess
+
+from distutils.version import LooseVersion
+from ansible import __version__
+from ansible import constants as C
+from ansible.errors import AnsibleParserError
+from ansible.module_utils._text import to_bytes, to_native, to_text
+
+if LooseVersion(__version__) >= LooseVersion('2.4'):
+    from ansible.plugins.vars import BaseVarsPlugin
+else:
+    BaseVarsPlugin = object
+
+from ansible.inventory.host import Host
+from ansible.inventory.group import Group
+from ansible.utils.vars import combine_vars
+
+FOUND = {}
+
+
+class VarsModule(BaseVarsPlugin):
+
+    def _should_i_run(self):
+        try:
+            subprocess.call([DECRYPT_TOOL], stdout=subprocess.PIPE)
+        except OSError as e:
+            if e.errno == errno.ENOENT:
+                self._display.warning("load_secret plugin cannot to find %s executable" % DECRYPT_TOOL)
+            return False
+        else:
+            return True
+
+    def _decrypt_file(self, file_path, entity):
+        self._display.display("Trying to decrypt %s" % file_path)
+        temp_plain_file=tempfile.NamedTemporaryFile(dir="./host_vars", prefix="%s-" % entity.name, suffix='-tmp-secrets')
+        exit_code = subprocess.call([DECRYPT_TOOL, "cat", file_path], stdout=temp_plain_file)
+        if exit_code > 0:
+            self._display.warning("Decryption failed with error %s for file %s" % (exit_code, file_path))
+        return (exit_code, temp_plain_file)
+
+
+    def get_vars(self, loader, path, entities, cache=True):
+
+        data = {}
+        if not self._should_i_run():
+            return data
+
+        ''' parses the inventory file '''
+        if not isinstance(entities, list):
+            entities = [entities]
+
+        super(VarsModule, self).get_vars(loader, path, entities)
+        for entity in entities:
+            if isinstance(entity, Host):
+                secrets_dir = os.path.realpath(os.path.join(os.getcwd(), '../../secret/build/'))
```

Comment:
Is it possible to make this configurable? For example, I store my Git repos locally as `~/code/nodejs-private/secrets` and `~/code/nodejs/build` 😬

Reply:
@maclover7 mmm not sure, I can have a look, where would you store the configuration though?

Reply:
Maybe an environment variable? Not sure. More just thinking out loud, since we can't guarantee that everyone will have the same folder layout on their machines